src/java.base/share/classes/javax/crac/Core.java

```diff
@@ -82,4 +82,13 @@ public static void checkpointRestore() throws
             throw newException;
         }
     }
+
+    /**
+     * Gets new arguments provided after restore.
```

Comment:
May be description could be extended. I think it's worth mentioning that the new arguments do not replace original arguments, it is application-level arguments and it's application's responsibility to parse and apply these additional arguments. An empty array is returned in case of no additional arguments are provided.

Reply:
I was trying to provide better docs without over-specifying current implementation and realized that the new functionality can be done without an API change. Please look at the latest state.

Reply:
New approach looks great. Thank you